Spanner/src/Connection/Grpc.php

```diff
@@ -455,6 +457,38 @@ public function createSession(array $args)
         ]);
     }
 
+    /**
+     * @return PromiseInterface
+     */
+    public function createSessionAsync(array $args)
+    {
+
+        $database = $this->pluck('database', $args);
+        $opts = $this->addResourcePrefixHeader([], $database);
+        $opts['credentialsWrapper'] = $this->credentialsWrapper;
+        $transport = $this->spannerClient->getTransport();
+
+        $request = new CreateSessionRequest();
+        $request->setDatabase($database);
```

Comment:
Could we use constructor injection for this?

```php
$request = new CreateSessionRequest([
    'database' => $database
]);
```

That was added after the clients were written, which is why it's not done in more cases.